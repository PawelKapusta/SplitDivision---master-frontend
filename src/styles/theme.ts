import { DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
    palette: {
        white: "#fff",
        black: "#000",
        background1: "#0F1624",
        gold: "#eaaf57",
    },
    breakpoints: {
        sm: "screen and (max-width: 640px)",
        md: "screen and (max-width: 768px)",
        lg: "screen and (max-width: 1024px)",
        xl: "screen and (max-width: 1280px)",
    },
};
