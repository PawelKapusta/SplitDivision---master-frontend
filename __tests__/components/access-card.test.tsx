import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import AccessCard from "@components/access-card/index";
import { mockTheme } from "../mocks/theme";

describe("access-card component", () => {
    it("renders correctly", () => {
        const { container } = render(
            <ThemeProvider theme={mockTheme}>
                <AccessCard imageSrc={"/path/image.png"}>
                    <div>Children</div>
                </AccessCard>
            </ThemeProvider>,
        );
        const image = screen.getByAltText("/path/image.png image");
        const children = screen.getByText("Children");
        expect(image).toBeInTheDocument();
        expect(children).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
