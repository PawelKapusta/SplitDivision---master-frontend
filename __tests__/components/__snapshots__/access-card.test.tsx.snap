// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`access-card component renders correctly 1`] = `
<div>
  <div
    class="sc-b975094-0 aSLkC"
  >
    <div
      class="sc-b975094-1 cwvPMC"
    >
      <div
        class="sc-b975094-2 juIojN"
      >
        <img
          alt="bitcoin-icon"
          class="sc-b975094-3 EKWit"
          src="/icons/bitcoin.svg"
        />
      </div>
      <img
        alt="/path/image.png image"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="1000"
        src="/_next/image?url=%2Fpath%2Fimage.png&w=2048&q=75"
        srcset="/_next/image?url=%2Fpath%2Fimage.png&w=1080&q=75 1x, /_next/image?url=%2Fpath%2Fimage.png&w=2048&q=75 2x"
        style="color: transparent;"
        width="1000"
      />
    </div>
    <div>
      Children
    </div>
  </div>
</div>
`;
